include/slang/ast/expressions/AssertionExpr.h

```diff
@@ -87,16 +119,21 @@ class SLANG_EXPORT AssertionExpr {
     /// Indicates whether the expression is invalid.
     bool bad() const { return kind == AssertionExprKind::Invalid; }
 
-    /// @returns true if this is a sequence expression that admits an empty match,
-    /// and false otherwise.
-    bool admitsEmpty() const;
+    // Checks that the sequence matches one of the following:
+    //   - admits empty matches
+    //   - accepts only empty matches
+    //   - admits no matches
+    bitmask<NondegeneracyStatus> checkNondegeneracy() const;
+
+    /// Computes possible clock ticks (delay) length of sequence under assertion expression
+    std::optional<SequenceRange> computeSequenceLength() const;
 
     static const AssertionExpr& bind(const syntax::SequenceExprSyntax& syntax,
                                      const ASTContext& context, bool allowDisable = false);
 
-    static const AssertionExpr& bind(const syntax::PropertyExprSyntax& syntax,
-                                     const ASTContext& context, bool allowDisable = false,
-                                     bool allowSeqAdmitEmpty = false);
+    static const AssertionExpr& bind(
+        const syntax::PropertyExprSyntax& syntax, const ASTContext& context,
+        bool allowDisable = false, const NondegeneracyFlags nondegFlags = NondegeneracyFlags::None);
```

Comment:
The `const` here does nothing -- since this is passed by value a copy will be made, which is fine, it's a small type.

Reply:
fixed